c/externs.h

```diff
@@ -45,6 +45,16 @@ off64_t lseek64(int,off64_t,int);
 #include <setjmp.h>
 #endif
 
+#ifndef NORETURN
+
+#if defined(__GNUC__) || defined(__clang__)
+#define NORETURN __attribute__((noreturn))
+#else
```

Comment:
Why is there no MSVC clause with `__declspec(noreturn)`? 
In https://github.com/cisco/ChezScheme/blob/master/BUILDING it says it is possible to build with MSVC.

Reply:
I took it out because I noticed that it was breaking the Travis Windows CI builds and I hadn't actually tested that piece so didn't want to leave it in if it broke the CI build.

When I get a chance I'll make arrangements to try building ChezScheme on Windows and test it out. If I'm able to get it working I'll push a commit adding it back in. Though in all fairness I'm wondering it's even necessary? If we can build CS on Windows fine without `___declspec(noreturn)` then maybe it's not necessary? 

One difficulty I noticed is that it appears that `__declspec` has to be put at the front of the function declarations while `__attribute__` needs to be put at the end, I'm not entirely sure how to resolve that using a macro but I'll do some research.

Reply:
You can add it at the front. It will work for both GCC and MSVC. @mflatt did that already in racket/ChezScheme. I always think that it's a good idea to provide compiler with our intentions (as developers). Even if it compiles without the `__declspec`, I would prefer if it's provided.

Reply:
I've moved to the front and confirm it compiles without issues on my Windows machine. Thanks for the suggestion!